src/internal/adapter/web/WebPresentationModeManager.ts

```diff
@@ -24,57 +24,47 @@ export class WebPresentationModeManager {
 
   set presentationMode(presentationMode: PresentationMode) {
     if (presentationMode === this._presentationMode) {
+      // Ignore if presentationMode did not change.
       return;
     }
 
-    this.prepareForPresentationModeChanges();
+    this.maybePrepareForPresentationModeChanges();
```

Comment:
Wouldn't it be better if we add the eventlisteners in the constructor already?

Reply:
Yes, that makes sense indeed.